core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java

```diff
@@ -110,6 +108,10 @@ public InnerHitsContext innerHits() {
         public void setChildInnerHits(Map<String, InnerHitsContext.BaseInnerHits> childInnerHits) {
             this.childInnerHits = new InnerHitsContext(childInnerHits);
         }
+
+        boolean doCountOnly() {
+            return from() < 0 || from() >= size() || size() <= 0;
+        }
```

Comment:
This does not make sense to me: fi there are 7 inner hits and from=2 and size=1, then we should not just count?